tests/avify4mtest.c

```diff
@@ -0,0 +1,196 @@
+            for (uint32_t y = 0; y < planeHeight; ++y) {
+                uint8_t * const row = image->yuvPlanes[plane] + y * image->yuvRowBytes[plane];
+                if (image->depth == 8) {
+                    memset(row, yuvValue, planeWidth);
+                } else {
+                    for (uint32_t x = 0; x < planeWidth; ++x) {
+                        ((uint16_t *)row)[x] = yuvValue;
+                    }
+                }
+            }
+        }
+    }
+    if (image->alphaPlane != NULL) {
+        assert(image->alphaRange == AVIF_RANGE_FULL);
+        const uint32_t alphaValue = (1 << image->depth) - 1;
+        for (uint32_t y = 0; y < image->height; ++y) {
+            uint8_t * const row = image->alphaPlane + y * image->alphaRowBytes;
+            if (image->depth == 8) {
+                memset(row, alphaValue, image->width);
+            } else {
+                for (uint32_t x = 0; x < image->width; ++x) {
+                    ((uint16_t *)row)[x] = alphaValue;
+                }
+            }
+        }
+    }
+}
+
+// Creates an image and encodes then decodes it as a y4m file.
+static avifBool encodeDecodeY4m(uint32_t width,
+                                uint32_t height,
+                                uint32_t depth,
+                                avifPixelFormat yuvFormat,
+                                avifRange yuvRange,
+                                avifBool createAlpha,
+                                const char filePath[])
+{
+    avifBool success = AVIF_FALSE;
+    avifImage * image = avifImageCreateEmpty();
+    avifImage * decoded = avifImageCreateEmpty();
+    if (image == NULL || decoded == NULL) {
+        printf("ERROR: avifImageCreate() failed\n");
+        goto cleanup;
+    }
+    image->width = width;
+    image->height = height;
+    image->depth = depth;
+    image->yuvFormat = yuvFormat;
+    image->yuvRange = yuvRange;
+    image->alphaRange = AVIF_RANGE_FULL; // Unused by y4mRead() and y4mWrite().
+    avifImageAllocatePlanes(image, createAlpha ? AVIF_PLANES_ALL : AVIF_PLANES_YUV);
+    fillPlanes(image);
+
+    if (!y4mWrite(filePath, image)) {
+        printf("ERROR: y4mWrite() failed\n");
+        goto cleanup;
+    }
+    if (!y4mRead(filePath, decoded, /*sourceTiming=*/NULL, /*iter=*/NULL)) {
+        printf("ERROR: y4mRead() failed\n");
+        goto cleanup;
+    }
+
+    if (!compareYUVA(image, decoded)) {
+        goto cleanup;
+    }
+
+    success = AVIF_TRUE;
+cleanup:
+    if (image != NULL) {
+        avifImageDestroy(image);
+    }
+    if (decoded != NULL) {
+        avifImageDestroy(decoded);
+    }
+    return success;
+}
+
+//------------------------------------------------------------------------------
+
+int main(int argc, char * argv[])
+{
+    if (argc < 2 || !strlen(argv[1])) {
+        fprintf(stderr, "Missing temporary file path argument\n");
+        return EXIT_FAILURE;
+    }
+    char filePath[100];
+    sprintf(filePath, "%s/avify4mtest.y4m", argv[1]);
```

Comment:
I looked into this. Among the alternatives suggested by the [tmpnam page](https://en.cppreference.com/w/c/io/tmpnam) on cppreference.com, I like the POSIX function [`tempnam()`](https://pubs.opengroup.org/onlinepubs/9699919799/functions/tempnam.html) the best. On Windows we can use [`_tempnam()`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/tempnam-wtempnam-tmpnam-wtmpnam?view=msvc-150).

Reply:
1. I would rather avoid non-portable solutions that would introduce platform-dependent defines.
2. Done.

Reply:
Using `tempnam()` allows us to run the same avify4mtest binary from multiple terminal windows at the same time. This is admittedly an uncommon use case. This kind of simple portability issue (where `tempname()` and `_tempname()` differ only in the function name) can be handled by using a macro or adding a wrapper function.